kubectl-pg/cmd/scale.go

```diff
@@ -0,0 +1,144 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	PostgresqlLister "github.com/zalando/postgres-operator/pkg/generated/clientset/versioned/typed/acid.zalan.do/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"log"
+	"strconv"
+)
+
+// scaleCmd represents the scale command
+var scaleCmd = &cobra.Command{
+	Use:   "scale",
+	Short: "Add/remove pods to a Postgres cluster",
+	Long: `Scales the postgres objects using cluster-name.
+Scaling to 0 leads to down time.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		clusterName, err := cmd.Flags().GetString("clusterName")
+		namespace, err := cmd.Flags().GetString("namespace")
+		if err != nil {
+			log.Fatal(err)
+		}
+		if len(args) > 0 {
+			numberofInstances, err := strconv.Atoi(args[0])
+			if err != nil {
+				log.Fatal(err)
+			}
+			scale(int32(numberofInstances), clusterName, namespace)
+		} else {
+			fmt.Println("Please enter number of instances to scale.")
+		}
+
+	},
+	Example: "kubectl pg scale [NUMBER-OF-INSTANCES] -c [CLUSTER-NAME] -n [NAMESPACE]",
+}
+
+func scale(numberOfInstances int32, clusterName string, namespace string) {
+	config := getConfig()
+	postgresConfig, err := PostgresqlLister.NewForConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	postgresql, err := postgresConfig.Postgresqls(namespace).Get(clusterName, metav1.GetOptions{})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if numberOfInstances == 0 {
+		fmt.Printf("Scaling to zero leads to down time. please type %s/%s and hit Enter this serves to confirm the action\n", namespace, clusterName)
+		confirmAction(clusterName, namespace)
+	}
+	minInstances, maxInstances := allowedMinMaxInstances(config)
+	if minInstances == -1 && maxInstances == -1 {
+		postgresql.Spec.NumberOfInstances = numberOfInstances
+	} else if numberOfInstances <= maxInstances && numberOfInstances>= minInstances {
+		postgresql.Spec.NumberOfInstances = numberOfInstances
+	} else if minInstances == -1 && numberOfInstances < postgresql.Spec.NumberOfInstances ||
+		 maxInstances == -1 && numberOfInstances > postgresql.Spec.NumberOfInstances{
+		postgresql.Spec.NumberOfInstances = numberOfInstances
+	} else {
+		log.Fatal("cannot scale to the provided instances as they don't adhere to MIN_INSTANCES and MAX_INSTANCES provided in configmap or operatorconfiguration")
+	}
+	postgresql.Kind = "postgresql"
```

Comment:
such constants are defined in the codebase: [Kind](https://github.com/zalando/postgres-operator/blob/31e568157b336592debbb37f2c44c1ca1769c00d/pkg/apis/acid.zalan.do/v1/crds.go#L11) / [ApiVersion](https://github.com/zalando/postgres-operator/blob/db72d82f14cc4a2178504efc899df8158f6e9a4d/pkg/apis/acid.zalan.do/v1/register.go#L13)